src/server/v2.0/handler/preheat.go

```diff
@@ -80,3 +86,128 @@ func convertProvidersToFrontend(backend []*provider.Metadata) (frontend []*model
 	}
 	return
 }
+
+// GetPolicy is Get a preheat policy
+func (api *preheatAPI) GetPolicy(ctx context.Context, params operation.GetPolicyParams) middleware.Responder {
+	var payload *models.PreheatPolicy
+	policy, err := api.preheatCtl.GetPolicy(ctx, params.PreheatPolicyID)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+
+	payload, err = convertPolicyToPayload(policy)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+	return operation.NewGetPolicyOK().WithPayload(payload)
+}
+
+// CreatePolicy is Create a preheat policy under a project
+func (api *preheatAPI) CreatePolicy(ctx context.Context, params operation.CreatePolicyParams) middleware.Responder {
+	policy, err := convertParamPolicyToModelPolicy(params.Policy)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+
+	_, err = api.preheatCtl.CreatePolicy(ctx, policy)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+	return operation.NewCreatePolicyCreated()
+}
+
+// UpdatePolicy is Update preheat policy
+func (api *preheatAPI) UpdatePolicy(ctx context.Context, params operation.UpdatePolicyParams) middleware.Responder {
+	policy, err := convertParamPolicyToModelPolicy(params.Policy)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+
+	err = api.preheatCtl.UpdatePolicy(ctx, policy)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+	return operation.NewUpdatePolicyOK()
+}
+
+// DeletePolicy is Delete a preheat policy
+func (api *preheatAPI) DeletePolicy(ctx context.Context, params operation.DeletePolicyParams) middleware.Responder {
+	err := api.preheatCtl.DeletePolicy(ctx, params.PreheatPolicyID)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+
+	return operation.NewDeleteInstanceOK()
+}
+
+// ListPolicies is List preheat policies
+func (api *preheatAPI) ListPolicies(ctx context.Context, params operation.ListPoliciesParams) middleware.Responder {
+	query, err := api.BuildQuery(ctx, params.Q, params.Page, params.PageSize)
+	if err != nil {
+		return api.SendError(ctx, err)
+	}
+
+	if query != nil {
+		query.Keywords["project_id"] = params.ProjectID
+	}
```

Comment:
query.Keywords["ProjectID"] = pro.ProjectID

Reply:
We need to double-check the property key in the keywords should be the column name in the DB schema, right?

Reply:
the column name in db schema is 'project_id'